crates/sui-types/src/messages_consensus.rs

```diff
@@ -28,6 +28,32 @@ pub struct ConsensusCommitPrologue {
     pub commit_timestamp_ms: CheckpointTimestamp,
 }
 
+#[derive(Debug, Hash, PartialEq, Eq, Clone, Serialize, Deserialize)]
+pub struct AuthenticatorStateCreate {
+    // note: epoch is not used, it exists just to give the transaction a guaranteed
+    // unique digest.
+    /// Epoch of the authenticator state create transaction
+    pub epoch: u64,
```

Comment:
Now that we do these in end of epoch, these types no longer belong to messages_consensus. Also we no longer need the epoch here.

Reply:
We do need the epoch, otherwise we'll have duplicated tx digests. (there is a comment above explaining this)

Moving to transactions.rs

Reply:
This is part of the EndOfEpoch tx now, which is guaranteed to have a unique tx digest since epoch is already part of it.